experimental/lib/Transforms/Speculation/HandshakeSpeculation.cpp

```diff
@@ -342,7 +345,26 @@ std::optional<Value> findControlInputToBB(Operation *op) {
   for (const BBArc &arc : bbToPredecessorArcs[bb]) {
     // Iterate the operands in the edge
     for (mlir::OpOperand *p : arc.edges) {
-      Value inEdge = p->get();
+      Operation *ctrlSignalFrom = p->getOwner();
+      // At first ctrlSignalFrom should be a CMerge.
+      // TODO: I think there is an easier way to find ControlMergeOp
+      // just like PlacementFinder::findSaveCommitPositions in PlacementFinder.cpp,
+      // which doesn't use BBToArcsMap but uses funcOp.getOps
```

Comment:
I can do this in this PR

Reply:
is there a big difference in complexity for [building the arcs and iterating over them] vs [using getOps]?

Reply:
Not much and that's for sure that getOps still requires iteration. So now I think it's okay to leave the code as is